proxy/sources/imgchest.py

```diff
@@ -0,0 +1,106 @@
+from typing import List, Optional, Dict
+
+from bs4 import BeautifulSoup
+from django.shortcuts import redirect
+from django.urls import re_path
+
+from proxy.source import ProxySource, SeriesPage, ChapterAPI, SeriesAPI, api_cache, get_wrapper
+
+
+class ImageChest(ProxySource):
+    def get_reader_prefix(self) -> str:
+        return "imgchest"
+
+    def shortcut_instantiator(self) -> List[re_path]:
+        def handler(_, album_hash):
+            slug = f"reader-{self.get_reader_prefix()}-chapter-page"
+            return redirect(slug, album_hash, "1", "1")
+
+        return [re_path(r"^p/(?P<album_hash>\w+)/$", handler)]
+
+    @api_cache(prefix="imgchest_api_dt", time=300)
+    def imgchest_common(self, meta_id: str) -> Optional[Dict]:
+        url = f"https://imgchest.com/p/{meta_id}"
+
+        resp = get_wrapper(url)
+        if resp.status_code != 200:
+            return None
+
+        soup = BeautifulSoup(resp.text, "html.parser")
+        page_elements = soup.find_all("meta", attrs={"name": "twitter:image"})
+        pages = [page["content"] for page in page_elements]
+
+        if pages.count(None) == len(pages):
+            # Could not retrieve content attribute from any image element.
+            return None
```

Comment:
We could probably raise a `ProxyException` that bubbles up to the user if we hit this case.